### PR TITLE
Add support in spin to set the flag store using only the store name

### PR DESCRIPTION
**Pull request by alvaro-shopify**
### WHY are these changes introduced?
The flag `--store` available when running the command `dev` allows the user to set the store used to install the app. The final requirement to install the app in the store is setting the complete `store hostname`.
The CLI supports either setting the  hostname or only the store name. In case the second option is used the CLI complete it with the domain to build the hostname. However this feature is not supported when the CLI is run against a spin instance.

### WHAT is this pull request doing?

- When the user only sets the store name for the flag the CLI checks the current environment and complete the store hostname properly

### How to test your changes?
- Run the `dev` command against a spin instance and use the flag `--store <only-store-name>`

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Review by matteodepalo — APPROVED** [review]

File: packages/cli-kit/src/string.ts [hunk 1]
```diff
@@ -39,15 +40,18 @@ export function capitalize(string: string) {
 }
 
 /**
- * Given a store, returns a valid store fqdn removing protocol and adding .myshopify.com domain
+ * Given a store, returns a valid store fqdn removing protocol and adding the proper domain in case is missing
  * @param store - Original store name provided by the user
  * @returns a valid store fqdn
  */
 export function normalizeStoreName(store: string) {
```
Comment (amcaplan) [inline packages/cli-kit/src/string.ts:47]: It feels like `normalizeStoreName` is moving beyond a `string` operation. Is there a better place for this to live?

File: packages/cli-kit/src/session.ts [hunk 2]
```diff
@@ -190,8 +190,12 @@ export async function ensureAuthenticated(
...
   const fqdn = await identityFqdn()
 
-  if (applications.adminApi?.storeFqdn) {
-    applications.adminApi.storeFqdn = normalizeStoreName(applications.adminApi.storeFqdn)
+  const previousStoreFqdn = applications.adminApi?.storeFqdn
+  if (previousStoreFqdn) {
+    const normalizedStoreName = await normalizeStoreName(previousStoreFqdn)
+    if (previousStoreFqdn === applications.adminApi?.storeFqdn) {
+      applications.adminApi.storeFqdn = normalizedStoreName
+    }
```
Comment (isaacroldan) [inline packages/cli-kit/src/session.ts:198]: do you need this `if`? it should be ok to always update the storeFqdn no?

File: packages/cli-kit/src/string.ts (hunk 1 above)
Comment (isaacroldan) [reply packages/cli-kit/src/string.ts:47]: agree

**Review by isaacroldan — APPROVED** [review]

File: packages/cli-kit/src/session.test.ts [hunk 3]
```diff
@@ -111,7 +110,7 @@ beforeAll(() => {
 })
 
 beforeEach(() => {
-  vi.mocked(identity).mockResolvedValue(fqdn)
+  vi.spyOn(fqdnModule, 'identity').mockResolvedValue(fqdn)
```
[the remark this answers is not in the document]
Comment (alvaro-shopify) [reply packages/cli-kit/src/session.test.ts:113]: I have moved the the method `normalizeStoreName` to the `fqdn` module. Previously in these tests the complet `fqdn` module was mocked so some tests were failing. 
I think that when you only need to mock one method from a module, it's better to do it in this ways so no collateral effects arise. 
In other tests we are using a different alternative:
```js
vi.mock('./environment/fqdn', async () => {
  const fqdn: any = await vi.importActual('./environment/fqdn')
  return {
    ...fqdn,
    identity: vi.fn(),
    },
  }
})
vi.mocked(identity).mockResolvedValue(fqdn)
```
but I think using the `spyOn` is less cumbersome.
@isaacroldan  what do you think about this?

**Pull request merged**